lib/src/material_color_picker.dart

```diff
@@ -170,9 +171,9 @@ class _MaterialColorPickerState extends State<MaterialColorPicker> {
 
   @override
   Widget build(BuildContext context) {
-    final listChildren = _isMainSelection || !widget.allowShades
-        ? _buildListMainColor(_colors)
-        : _buildListShadesColor(_mainColor);
+    final listChildren = _isMainSelection ?? false || !widget.allowShades
```

Comment:
You'll need some parenthesis here. `??` has a lower precedence than `||`, so this is evaluated as `_isMainSelection ?? (false || !widget.allowShades)`. I think it should be `(_isMainSelection ?? false) || !widget.allowShades`.

EDIT: I also added another suggestion for making `_isMainSelection` not nullable, so if you do that this is no longer needed.